### PR TITLE
#94 checking for null project service

### PR DESCRIPTION
**Pull request by alesyalis**

### CODE REVIEW

File: Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public IQueryable<Candidate> GetByProject(Guid projectId)
 
         public IQueryable<Candidate> GetCandidatesPassedTestByProject(Guid projectId)
         {
-            return GetCollection().AsQueryable().Where(x => x.ProjectResults.Any(p => p.ProjectId == projectId && p.Status == Status.Test));
+            return GetCollection().AsQueryable().Where(x => x.ProjectResults.Any(p => p.ProjectId == projectId && p.Status == Status.Test && p.Feedbacks.Any(x => x.Rating != 0)));
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs:28]: You check what any of feedbacks should be not 0. Is it correct?? You should check that test feedback not 0

File: Recrutify/Recrutify.Host/Controllers/OData/CandidatesController.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ public CandidatesController(ICandidateService candidateService)
 
         [EnableQuery(
             HandleNullPropagation = HandleNullPropagationOption.False,
-            MaxAnyAllExpressionDepth = 2,
+            MaxAnyAllExpressionDepth = 3,
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/OData/CandidatesController.cs:26]: Why 3? It'a not necessary if it doesn't help with sorting

File: Recrutify/Recrutify.Host/Controllers/ProjectsController.cs [hunk 3]
```diff
@@ -84,13 +85,15 @@ public async Task<ActionResult> DeleteAsync([FromRoute] Guid id, CancellationTok
...
         public async Task<ActionResult<ProjectDTO>> GetByIdAsync([FromRoute] Guid id)
         {
-            var project = await _projectService.GetAsync(id);
-            if (project == null)
+            try
+            {
+                var project = await _projectService.GetAsync(id);
+                return Ok(project);
+            }
+            catch (NotFoundException)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/ProjectsController.cs:93]: You should catch not only here, but in all places where you call project service get method - in all controllers where this exception can be thrown

File: Recrutify/Recrutify.Host/Controllers/OData/CandidatesController.cs (hunk 2 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Controllers/OData/CandidatesController.cs]: done

File: Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs (hunk 1 above)
Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/CandidateRepository.cs]: done

File: Recrutify/Recrutify.Host/Controllers/ProjectsController.cs (hunk 3 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Controllers/ProjectsController.cs:93]: that's all

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**